File: MicroservicioVentas/src/main/java/com/Conjunta/MicroservicioVentas/Service/VentasService.java
package com.Conjunta.MicroservicioVentas.Service;

import com.Conjunta.MicroservicioVentas.DTO.VentasDTO;
import com.Conjunta.MicroservicioVentas.Model.Ventas;

import java.util.List;

public interface VentasService {

    // Crear una nueva venta
    Ventas crearVenta(VentasDTO ventasDTO);

    // Obtener todas las ventas
    List<Ventas> obtenerTodasLasVentas();

    // Obtener una venta por su ID
    Ventas obtenerVentaPorId(Long id);

    // Actualizar una venta existente
    Ventas actualizarVenta(Long id, VentasDTO ventasDTO);

    // Eliminar una venta por su ID
    boolean eliminarVenta(Long id);
}

File: Frontend_Grupo2/src/app/service/ventas.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Ventas } from '../models/ventas'; // Ajusta la ruta según tu estructura

@Injectable({
  providedIn: 'root'
})
export class VentasService {

  private apiUrl = 'http://localhost:8082/api/ventas'; // URL base de la API

  // Opciones para la solicitud HTTP
  private httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json'
    })
  };

  constructor(private http: HttpClient) { }

  // Crear una nueva venta
  public crearVenta(venta: Ventas): Observable<Ventas> {
    return this.http.post<Ventas>(this.apiUrl, venta, this.httpOptions);
  }

  // Obtener todas las ventas
  public obtenerTodasLasVentas(): Observable<Ventas[]> {
    return this.http.get<Ventas[]>(this.apiUrl);
  }

  // Obtener una venta por su ID
  public obtenerVentaPorId(id: number): Observable<Ventas> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.get<Ventas>(url);
  }

  // Actualizar una venta existente
  public actualizarVenta(id: number, venta: Ventas): Observable<Ventas> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.put<Ventas>(url, venta, this.httpOptions);
  }

  // Eliminar una venta por su ID
  public eliminarVenta(id: number): Observable<void> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.delete<void>(url);
  }
}
